cpp/scripts/include_checker.py

```diff
@@ -22,32 +22,81 @@
 import io
 from functools import reduce
 import operator
-
+import dataclasses
+import typing
 
 # file names could (in theory) contain simple white-space
 IncludeRegex = re.compile(r"(\s*#include\s*)([\"<])([\S ]+)([\">])")
+PragmaRegex = re.compile(r"^ *\#pragma\s+once *$")
+UsingNamespaceRegex = re.compile(r"^ *\#pragma\s+once *$")
```

Comment:
In the PR description I did mention that this PR does add an additional check for bad `#pragma once` usages (not the first line of code).

I was planning on adding a check for poor `using namespace XXX;` but decided against it since there were very little places that benefited and there was already an issue created to address this.

Line 31 should have been removed.